components/camel-hazelcast/src/main/java/org/apache/camel/component/hazelcast/topic/HazelcastTopicProducer.java

```diff
@@ -41,19 +42,18 @@ public HazelcastTopicProducer(HazelcastInstance hazelcastInstance, HazelcastDefa
     }
 
     public void process(Exchange exchange) throws Exception {
-        final int operation = lookupOperationNumber(exchange);
+        final HazelcastOperation operation = lookupOperation(exchange);
 
-        switch (operation) {
-        case -1:
-            // default operation to publish
-        case HazelcastConstants.PUBLISH_OPERATION:
+        switch ((operation != null) ? operation : HazelcastOperation.PUBLISH) {
```

Comment:
If lookupOperation cannot be null, how do you deal with old _-1_ value ? Basically this current _null_ value is the former _-1_ value.
The defaultOperation for Queue is _ADD_. But the defaultOperation for Topic is _PUBLISH_.